lib/loadObj.js

```diff
@@ -195,24 +296,196 @@ function loadObj(objPath, options) {
+            isWindingCorrect = checkWindingCorrect(positions[0], positions[1], positions[2], faceNormal);
+        }
+
+        if (vertices.length === 3) {
+
+            if (uvs) {
+                u1 = uvs[0];
+                u2 = uvs[1];
+                u3 = uvs[2];
+            }
+
+            if (normals) {
+                n1 = normals[0];
+                n2 = normals[1];
+                n3 = normals[2];
+            }
+
+            var index1 = addVertex(vertices[0], positions[0], u1, n1);
+            var index2 = addVertex(vertices[1], positions[1], u2, n2);
+            var index3 = addVertex(vertices[2], positions[2], u3, n3);
+
+            addTriangle(index1, index2, index3, isWindingCorrect);
+        } else { // Triangulate if the face is not a triangle
+            var positions3D = [];
+            var vertexIndices = [];
+
+            var i;
+            for (i = 0; i < vertices.length; ++i) {
+
+                var u = (defined(uvs)) ? uvs[i] : undefined;
+                var n = (defined(normals)) ? normals[i] : undefined;
+
+                var index = addVertex(vertices[i], positions[i], u, n);
+                vertexIndices.push(index);
+
+                // Collect the vertex positions as 3D points
+                positions3D.push(get3DPoint(positions[i], new Cartesian3()));
+            }
+
+            var positions2D = projectTo2D(positions3D);
+
+            if (isConvex(positions2D)) {
+                for (i=1; i < vertices.length-1; ++i) {
+                    addTriangle(vertexIndices[0], vertexIndices[i], vertexIndices[i+1], isWindingCorrect);
+                }
+            } else {
+                // Since the projection doesn't preserve winding order, reverse the order of
+                // the vertices before triangulating to enforce counter clockwise.
+                var projectedWindingOrder = PolygonPipeline.computeWindingOrder2D(positions2D);
+                if (projectedWindingOrder === WindingOrder.CLOCKWISE) {
+                    positions2D.reverse();
+                }
+
+                // Use an ear-clipping algorithm to triangulate
+                var positionIndices = PolygonPipeline.triangulate(positions2D);
+                for (i = 0; i < positionIndices.length-2; i += 3) {
+                    addTriangle(vertexIndices[positionIndices[i]], vertexIndices[positionIndices[i+1]], vertexIndices[positionIndices[i+2]], isWindingCorrect);
+                }
+            }
+        }
+    }
+
+    function trimEmpty(entry) { return entry.trim() !== ''; }
+
+    // Parse a line of face data.
+    // Because face data can span multiple lines, attribute data is buffered in parallel lists
+    // faceVertices : names of a vertices for caching
+    // facePosition : indices into position array
+    // faceUvs : indices into uv array
+    // faceNormals : indices into normal array
+    function parseFaceLine(line, hasUvs, hasNormals) {
```

Comment:
Yeah, as far as I can tell, you have to just iterate through the string and grab attributes manually if there are n of them -- capture groups don't work with n elements.

As for getting rid of parseFaceLine, this will create a bunch of repeated code in each of the face cases --is that ok?